task/todoist.go

```diff
@@ -0,0 +1,31 @@
+package task
+
+import (
+	"github.com/Xuanwo/navvy"
+)
+
+// Todoist will holding a todo list for tasks.
+type Todoist interface {
+	AddTODOs(...func(Todoist) navvy.Task)
+	NextTODO() func(Todoist) navvy.Task
+}
+
+// Todo is the struct which inplements Todoist.
+type Todo struct {
+	v []func(Todoist) navvy.Task
+}
+
+// NextTODO will return next task to do.
+func (o *Todo) NextTODO() func(Todoist) navvy.Task {
+	if len(o.v) == 0 {
+		return nil
+	}
+	fn := o.v[0]
+	o.v = o.v[1:]
```

Comment:
We'd better add `o.v[0] = nil` here to do the release job, like what you did in navvy/pool.go L162
```
w = p.workers[l]
p.workers[l] = nil
p.workers = p.workers[:l]
```